cryptid_anchor/programs/middleware/check_recipient/src/lib.rs

```diff
@@ -0,0 +1,142 @@
+extern crate core;
+
+use anchor_lang::prelude::*;
+use cryptid_anchor::cpi::accounts::ApproveExecution;
+use cryptid_anchor::program::CryptidAnchor;
+use cryptid_anchor::state::transaction_account::TransactionAccount;
+
+declare_id!("midcHDoZsxvMmNtUr8howe8MWFrJeHHPbAyJF1nHvyf");
```

Comment:
you grinded for `midc` here?

Reply:
yes - we can regrind for something nicer if we like. I just wanted something temporarily easy to recognise.